lib/testutils/mockresolver/resolver.go

```diff
@@ -9,17 +9,17 @@ import (
 // MockResolver implements netext.Resolver, and allows changing the host
 // mapping at runtime.
 type MockResolver struct {
-	m        sync.RWMutex
-	hosts    map[string][]net.IP
-	fallback func(host string) ([]net.IP, error)
+	m           sync.RWMutex
+	hosts       map[string][]net.IP
+	ResolveHook func(host string, result []net.IP)
```

Comment:
nit: I'd suggest naming this `LookupIPAllHock` since `MockResolver` has also `LookupIP` and there we don't execute the hock after the name resolving.

Reply:
But `LookupIP` calls `LookupIPAll`, so the hook would run either way.